### PR TITLE
feat: implement proposal screen UI and integrate with business logic

### PR DESCRIPTION
**Pull request by nbetsaif**
## Scope

[Ticket](https://github.com/hypha-dao/hypha-wallet/issues/289)

Implemented the UI for the Proposals screen, integrated it with the existing business logic, and added error handling to improve user experience. 
`Note :` The DAO image and creator image are currently static placeholders and will be updated once the image retrieval logic is implemented.

### Light Mode
Simulator Screenshot - iPhone SE (3rd generation) - 2024-06-19 at 12 17 51

### Dark Mode
Simulator Screenshot - iPhone SE (3rd generation) - 2024-06-19 at 12 17 51

### Small Device
Simulator Screenshot - iPhone SE (3rd generation) - 2024-06-19 at 12 17 51

### Error Case
Simulator Screenshot - iPhone SE (3rd generation) - 2024-06-19 at 12 17 51

### CODE REVIEW

File: lib/core/extension/proposal_model_extensions.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    if (minutes > 0) {
+      parts.add('$minutes minute${minutes > 1 ? 's' : ''}');
+    }
+
+    return parts.isNotEmpty ? 'Expiring in ${parts.join(', ')}' : 'Expired';
+  }
+  double quorumToPercent() => quorum==null?0:quorum!*.01;
+  double unityToPercent() => unity==null?0:unity!*.01;
+}
+
```
Comment (n13) [inline lib/core/extension/proposal_model_extensions.dart:31]: I love using extensions for things like this 🙏

File: lib/ui/home_page/components/home_view.dart [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:flutter_svg/flutter_svg.dart';
+import 'package:get/get.dart' as GetX;
```
Comment (n13) [inline lib/ui/home_page/components/home_view.dart:4]: I see this is a little odd in the code - sometimes get is imported as GetX, sometimes now. Also prior to this change, so this is an older issue.

**Review by n13 — APPROVED** [review]
Looking great! 

Please merge at your convenience.

File: lib/ui/home_page/components/home_view.dart (hunk 2 above)
Comment (nbetsaif) [reply lib/ui/home_page/components/home_view.dart:4]: Thank you for pointing that out. I see the inconsistency with the import aliasing.If you have a preferred convention, like using **GetX** or keeping it as **get**, let me know, and I'll apply it across the code.

**Comment by n13** [comment]
It’s fine as it is On 27 Aug 2024, at 18:01, Nbet Saif ***@***.***> wrote:﻿
Merged #290 into main.

**Pull request merged**